File: Dot examples/ASP Book Examples/Book Examples/Chapter29/VB/App_Code/ItemRotator.vb
Imports System
Imports System.Collections
Imports System.Web.UI
Imports System.Web.UI.WebControls
Imports System.ComponentModel

Namespace myControls
    <ParseChildren(True, "Items")> _
    Public Class ItemRotator
        Inherits CompositeControl
        Private _items As ArrayList = New ArrayList()

        <Browsable(False)> _
        Public ReadOnly Property Items() As ArrayList
            Get
                Return _items
            End Get
        End Property

        Protected Overrides Sub CreateChildControls()
            Dim rnd As Random = New Random()
            Dim index As Integer = rnd.Next(_items.Count)
            Dim item As Control = CType(_items(index), Control)
            Me.Controls.Add(item)
        End Sub
    End Class

    Public Class Item
        Inherits Control

    End Class
End Namespace

File: Dot examples/ASP Book Examples/Book Examples/Chapter29/CS/App_Code/ItemRotator.cs
using System;
using System.Collections;
using System.Web.UI;
using System.Web.UI.WebControls;
using System.ComponentModel;

namespace myControls
{
    [ParseChildren(true, "Items")]
    public class ItemRotator : CompositeControl
    {
        private ArrayList _items = new ArrayList();

        [Browsable(false)]
        public ArrayList Items
        {
            get { return _items; }
        }

        protected override void CreateChildControls()
        {
            Random rnd = new Random();
            int index = rnd.Next(_items.Count);
            Control item = (Control)_items[index];
            this.Controls.Add(item);
        }
    }

    public class Item : Control
    {

    }
}